cli/internal/prune/prune.go

```diff
@@ -184,15 +185,44 @@ func (p *prune) prune(opts *turbostate.PrunePayload) error {
 		return errors.Wrap(err, "Failed to flush pruned lockfile")
 	}
 
-	if fs.FileExists(".gitignore") {
-		if err := fs.CopyFile(&fs.LstatCachedFile{Path: p.base.RepoRoot.UntypedJoin(".gitignore")}, fullDir.UntypedJoin(".gitignore").ToStringDuringMigration()); err != nil {
-			return errors.Wrap(err, "failed to copy root .gitignore")
-		}
+	// to reduce syscalls we just try to copy immediately. if it fails
+	// when getting mode, then continue because the file doesn't exist
+	err = fs.CopyFile(&fs.LstatCachedFile{Path: p.base.RepoRoot.UntypedJoin(".gitignore")}, fullDir.UntypedJoin(".gitignore").ToStringDuringMigration())
```

Comment:
Aside: I'd love to get `CopyFile` into a better place (not that it makes any sense to do this pre-Rust). At this point it's _only_ used in `prune`, and one teeny tiny place in `run_state.go`.

This used to be an important code path (enough so that `LstatCachedFile` as a perf optimization was worth it), but cache-to-tar has made it irrelevant.